internal/pkg/ecs/run_task_request.go

```diff
@@ -260,18 +269,36 @@ func containerInformation(taskDef *awsecs.TaskDefinition, containerName string)
 }
 
 // This function will format a map to a string as "key1=value1,key2=value2,key3=value3".
-func fmtStringMapToString(m map[string]string) string {
-	var output []string
-
-	// Sort the map so that `output` is consistent and the unit test won't be flaky.
+// Much of the complexity here comes from the two levels of escaping going on:
+// 1. we are outputting a command to be copied and pasted into a shell, so we need to shell-escape the output.
+// 2. the pflag library parses StringToString args as csv, so we csv escape the individual key/value pairs.
+func fmtStringMapToString(m map[string]string) (string, error) {
+	// Sort the map so that the output is consistent and the unit test won't be flaky.
 	keys := make([]string, 0, len(m))
 	for k := range m {
 		keys = append(keys, k)
 	}
 	sort.Strings(keys)
 
+	// write the key=value pairs as csv fields, which is what
+	// pflag expects to read in.
+	// This will escape internal double quotes and commas.
+	// We then need to trim the trailing newline that the csv writer adds.
+	var output []string
 	for _, k := range keys {
-		output = append(output, fmt.Sprintf(`%s="%v"`, k, m[k]))
+		output = append(output, fmt.Sprintf("%s=%s", k, m[k]))
+	}
+	buf := new(strings.Builder)
+	w := csv.NewWriter(buf)
+	err := w.Write(output)
+	if err != nil {
+		return "", err
 	}
-	return strings.Join(output, ",")
+	w.Flush()
+	final := strings.TrimSuffix(buf.String(), "\n")
+
+	// Then for shell escaping, wrap the entire argument in single quotes
+	// and escape any internal single quotes.
+	final = strings.ReplaceAll(final, `'`, `'\''`)
+	return `'` + final + `'`, nil
```

Comment:
nit: 
can we move this to a separate function

```go
func shellQuote(s string) string { 
    return `'` + strings.ReplaceAll(s, `'`, `'\''`) + `'`
}
```

Reply:
sure, done